cmd/metal-api/internal/service/ip-service.go

```diff
@@ -187,36 +195,17 @@ func (ir ipResource) freeIP(request *restful.Request, response *restful.Response
 	if checkError(request, response, utils.CurrentFuncName(), err) {
 		return
 	}
-
-	err = validateIPDelete(ip)
-	if checkError(request, response, utils.CurrentFuncName(), err) {
+	if checkError(request, response, utils.CurrentFuncName(), ir.actor.releaseIP(*ip)) {
 		return
 	}
 
-	err = ir.ipamer.ReleaseIP(*ip)
-	if checkError(request, response, utils.CurrentFuncName(), err) {
-		return
-	}
-
-	err = ir.ds.DeleteIP(ip)
-	if checkError(request, response, utils.CurrentFuncName(), err) {
-		return
-	}
 	err = response.WriteHeaderAndEntity(http.StatusOK, v1.NewIPResponse(ip))
 	if err != nil {
 		zapup.MustRootLogger().Error("Failed to send response", zap.Error(err))
 		return
 	}
 }
 
-func validateIPDelete(ip *metal.IP) error {
```

Comment:
I think the safer variant would be to leave this validation in the web service and not move it to the ip releaser. But maybe this is just because I do not know what this validation is actually meant for.

Reply:
You're right. the old implementation returned an error the the caller if the validation failed. My new implementation always returns OK but in the async code the validation could fail. i'll change this